rule/time_date.go

```diff
@@ -66,14 +75,44 @@ func (w lintTimeDate) Visit(n ast.Node) ast.Visitor {
 	}
 
 	// The last argument is a timezone, there is no need to check it, also it has a different type
-	for pos, arg := range ce.Args[:timeDateArity-1] {
+	for pos, realArg := range ce.Args[:timeDateArity-1] {
+		arg := realArg
+		var isNegative bool
+		ua, found := arg.(*ast.UnaryExpr)
+		if found {
+			// this is the argument we want to check
+			arg = ua.X
+			if ua.Op == token.SUB {
+				isNegative = true
+			}
+		}
+
 		bl, ok := arg.(*ast.BasicLit)
 		if !ok {
 			continue
 		}
+		fieldName := timeDateArgumentNames[pos]
 
-		replacedValue, err := parseDecimalInteger(bl)
+		parsedValue, err := parseDecimalInteger(bl)
 		if err == nil {
+			if isNegative {
+				parsedValue = -parsedValue
+			}
+
+			boundaries, found := timeDateArgumentBoundaries[fieldName]
+			if found {
+				if parsedValue < boundaries[0] || parsedValue > boundaries[1] {
```

Comment:
```suggestion
				min := boundaries[0]
				max := boundaries[1]
				if parsedValue < min || parsedValue > max {
```

(then in the failure msg we can also use `min` and `max`)